angularSTEP/src/app/upload-recipe/upload-recipe.component.html

```diff
@@ -1,93 +1,152 @@
+          </div>
+          <button [disabled]="toolsArray.length === 1" mat-raised-button color="primary" (click)="toolsArray.removeAt(i)">Delete</button>
+        </div>
+        <div>
+          <button mat-raised-button color="primary" [disabled]="!toolsArray.valid || checkForBlanks(toolsArray.value)" (click)="addToolsField()">Add Another</button>
+        </div>
+      </div>
+      <div class="control-buttons">
+        <button mat-raised-button color="primary" matStepperPrevious type="button">Back</button>
+        <button mat-raised-button color="primary" [disabled]="!toolsArray.valid || checkForBlanks(toolsArray.value)" matStepperNext type="button">Next</button>
+      </div>
+    </form>
+  </mat-step>
+
+  <mat-step [stepControl]="instructionsFormGroup">
+    <form [formGroup]="instructionsFormGroup" (ngSubmit)="onSubmit()" class ="form-array-formatting">
+      <ng-template matStepLabel>Instructions</ng-template>
+      <div formArrayName="instructionsArray">
+        <div *ngFor="let instruction of instructionsArray.controls; let i=index" class="array-item">
+          <mat-label>{{i + 1}}.</mat-label>          
+          <div class="input-error">
+            <input autocomplete="off" matInput type="text" [formControlName]="i">
+            <mat-error *ngIf="(instructionsArray.at(i).invalid || (removeNumsAndSymbolsAtStart(instructionsArray.at(i).value) === '')) && instructionsArray.at(i).touched">Please enter a valid item.</mat-error>
+          </div>
+          <button [disabled]="instructionsArray.length === 1" mat-raised-button color="primary" (click)="instructionsArray.removeAt(i)">Delete</button>
+        </div>
+        <div>
+          <button mat-raised-button color="primary" [disabled]="!instructionsArray.valid || checkInstructions(instructionsArray.value)" (click)="addInstructionsField()">Add Another</button>
+        </div>
+      </div>
+      <div class="control-buttons">
+        <button mat-raised-button color="primary" matStepperPrevious type="button">Back</button>
+        <button mat-raised-button color="primary" [disabled]="!instructionsArray.valid || checkInstructions(instructionsArray.value)" matStepperNext type="button">Next</button>
+      </div>
+    </form>
+  </mat-step>
+
+  <mat-step [stepControl]="extraFormGroup" [optional]="true">
+    <form [formGroup]="extraFormGroup" (ngSubmit)="onSubmit()" class ="form-formatting">
+      <ng-template matStepLabel>Extra Information</ng-template>
+      <mat-form-field>
+        <mat-label>Extra Information</mat-label>
+        <textarea autocomplete="off" matInput placeholder="Ex. It was a crisp autumn day when I suddenly felt inspired to take a trip down memory lane and cook..." formControlName="extraInfo"></textarea>
+        <mat-hint>Write anything else you may want to include about your recipe.</mat-hint>
+
+        <mat-form-field class="example-chip-list">
```

Comment:
line 118 to 142 of the upload-recipe.html is what I wrote everything else was added with the pull onto this branch